File: C/2020/56限/LatterPeriod/class_10.cpp
//  class_10.cpp
//  cl  /EHsc  class_10.cpp
#include <iostream>
class Hoge {
public:
    void p();
};
class Foo : public Hoge {
public:
    void p();
};
//
void Hoge::p() {
    std::cout << "Hoge::p" << std::endl;
}
void Foo::p() {
    std::cout << "Foo::p" << std::endl;
    Hoge::p();
}

int main() {
    //
    Hoge h;
    h.p();
    Foo f;
    f.p();

    return 0;
}

File: GameServer/2026/Sat34/class10.php
<?php  // class10.php
declare(strict_types=1);

readonly class Hoge {
    public string $name;

    public function __construct(string $name) {
        $this->name = $name;
    }
}

$obj = new Hoge("hogera");
var_dump($obj->name);

// Fatal error: Uncaught Error: Cannot modify readonly property Hoge::$name in
// $obj->name = "foo";

